lib/ash.ex

```diff
@@ -293,6 +293,12 @@ defmodule Ash do
                              ]},
                           doc:
                             "The fields to upsert. If not set, the action's upsert_fields is used, and if that is not set, then any fields not being set to defaults are written."
+                        ],
+                        upsert_condition: [
+                          # TODO: Better type?
```

Comment:
Unfortunately, `:any` is all we can really do. At some point in the future we will refactor expressions to live in `%Ash.Expr{}` so we can expect that struct, and make pattern matching much better in general.